### PR TITLE
[metrics-1] added basic metrics

### PR DESCRIPTION
**Pull request by CommanderStorm**
Adds these mertics:
- how often each detail (`/api/get/`) has been requested (2 Categories: Bot Traffic/Human Traffic)
- how often each detail (`/api/legacy_redirect/`) has been requested
- how long geting map images for the preview API took

From #440 

## Proposed Changes (include Screenshots if possible)

added the following metrics:
- `navigatum_APPNAME_incoming_requests` (labels: endpoint, method, status): the total number of HTTP requests handled by the actix HttpServer.
- `navigatum_APPNAME_response_code` (labels: endpoint, method, statuscode, type): Response codes of all HTTP requests handled by the actix HttpServer.
- `navigatum_APPNAME_response_time` (labels: endpoint, method, status): Total the request duration of all HTTP requests handled by the actix HttpServer.

## How to test this PR

1. Launch one of the servers
2. Do some requests
3. ```bash
   curl http://localhost:8080/metrics
   ```
   (note that for example a browser does not work, as these are not meant to be consumed by a browser)

## How has this been tested?

- curl request

## Checklist:

- [x] I have updated the documentation / No need to update the documentation
- [x] I have run the linter

### CODE REVIEW

**Comment by CommanderStorm** [comment]
This opens us up to a trivial DOS possiblity ⇒ is thus unaceptable
See https://github.com/nlopes/actix-web-prom/pull/46 for further details

This gives me the motivation needed to look if migrating to rocket and https://github.com/sd2k/rocket_prometheus is a better plan of action

**Comment by CommanderStorm** [comment]
I have checked out [rocket](https://rocket.rs/) and the features.
While they are really nice and some of the notation is way better, this comes at a cost:
This notation is achieved via procedural macros, which don't work sooo great with my IDE ⇒ not an option.

I have checked if this is really exploitable and have come to this conclusion:
While this is definitely exploitable, the effect is very limited.
The main reason is, that we filter URLs in our central ingress ⇒ employability is limited.

I have looked into, if we can provide a patch to [nlopes/actix-web-prom](https://github.com/nlopes/actix-web-prom) or [atomix-team/actix-web-prometheus](https://github.com/atomix-team/actix-web-prometheus): This is not possible due to the architecture of actix (what routes are available is not exposed to middlewares as far as I can tell)

Furthermore, the impact of this is, that Prometheus gets taken offline ⇒ this is picked up by Grafana ⇒ we are alerted and could fix this
Impact is only limited to one of our supporting services ⇒ not production

Thus, while not being ideal, I think this is a barely acceptabe risk (I am obviously not happy, but this will work)

File: server/calendar/src/main.rs [hunk 1]
```diff
@@ -29,6 +31,18 @@ async fn main() -> std::io::Result<()> {
     actix_rt::spawn(async move {
         scraping::continous_scraping::start_scraping(cloned_last_sync).await;
     });
+
+    // metrics
+    let labels = HashMap::from([(
+        "revision".to_string(),
+        std::env::var("GIT_COMMIT_SHA").unwrap_or("development".to_string()),
+    )]);
+    let prometheus = PrometheusMetricsBuilder::new("naviatum_calendar")
```
Comment (octycs) [inline server/calendar/src/main.rs:40]:
```suggestion
    let prometheus = PrometheusMetricsBuilder::new("navigatum_calendar")
```

File: server/main-api/src/main.rs [hunk 2]
```diff
@@ -24,6 +25,16 @@ async fn health_status_handler() -> HttpResponse {
 #[actix_web::main]
 async fn main() -> std::io::Result<()> {
     env_logger::init_from_env(env_logger::Env::default().default_filter_or("info"));
+    // metrics
+    let labels = HashMap::from([(
+        "revision".to_string(),
+        std::env::var("GIT_COMMIT_SHA").unwrap_or("development".to_string()),
+    )]);
+    let prometheus = PrometheusMetricsBuilder::new("naviatum_mainapi")
```
Comment (octycs) [inline server/main-api/src/main.rs:33]:
```suggestion
    let prometheus = PrometheusMetricsBuilder::new("navigatum_mainapi")
```

File: server/feedback/src/main.rs [hunk 3]
```diff
@@ -53,6 +55,17 @@ async fn main() -> std::io::Result<()> {
         .finish()
         .unwrap();
 
+    // metrics
+    let labels = HashMap::from([(
+        "revision".to_string(),
+        std::env::var("GIT_COMMIT_SHA").unwrap_or("development".to_string()),
+    )]);
+    let prometheus = PrometheusMetricsBuilder::new("naviatum_feedback")
```
Comment (octycs) [inline server/feedback/src/main.rs:63]:
```suggestion
    let prometheus = PrometheusMetricsBuilder::new("navigatum_feedback")
```

**Review by octycs — APPROVED** [review]
Just a typo but apart from that seems all good. If I understand [this](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) correctly, including the `Cargo.lock` files is better. I was not sure about that.

Since we do not store permanent data in any service that cannot be reretrived, I only see a risk of downtime but no data loss in the case someone tries to spam our metrics until storage space runs out.

**Pull request merged**